test/benchmark/l4bench.go

```diff
@@ -0,0 +1,194 @@
+		PerfLogWriter.Reset()
+	}, 500)
+
+	createContainers := func() {
+		vm.ContainerCreate(helpers.Client, helpers.NetperfImage, helpers.CiliumDockerNetwork, "-l id.client")
+		vm.ContainerCreate(helpers.Server, helpers.NetperfImage, helpers.CiliumDockerNetwork, "-l id.server")
+		vm.PolicyDelAll()
+		vm.WaitEndpointsReady()
+		err := helpers.WithTimeout(func() bool {
+			if data, _ := vm.GetEndpointsNames(); len(data) < 2 {
+				logger.Info("Waiting for endpoints to be ready")
+				return false
+			}
+			return true
+		}, "Endpoints are not ready", &helpers.TimeoutConfig{Timeout: 150})
+		Expect(err).Should(BeNil())
+	}
+
+	removeContainers := func(containerName string) {
+		By(fmt.Sprintf("removing container %s", containerName))
+		res := vm.ContainerRm(containerName)
+		Expect(res.WasSuccessful()).Should(BeTrue())
+	}
+
+	deleteContainers := func() {
+		removeContainers(helpers.Client)
+		removeContainers(helpers.Server)
+	}
+
+	BeforeEach(func() {
+	}, 500)
+
+	Context("Benchmark Netperf Tests Container2Container", func() {
+		BeforeAll(func() {
+			createContainers()
+		})
+
+		AfterAll(func() {
+			deleteContainers()
+		})
+
+		ClientServerCheck := func(serverIPv6 string, serverIP string) {
+			By("checking %s can ping to %s IPv6", helpers.Client, helpers.Server)
+			res := vm.ContainerExec(helpers.Client, helpers.Ping6(serverIPv6))
+			res.ExpectSuccess()
+
+			By("checking %s can ping to %s IPv4", helpers.Client, helpers.Server)
+			res = vm.ContainerExec(helpers.Client, helpers.Ping(serverIP))
+			res.ExpectSuccess()
+		}
+
+		SuperNetperfRRIPv4 := func(client bool, num int, serverIP string) {
+			var res *helpers.CmdRes
+
+			By("super_netperf to %s from %s (should succeed)", helpers.Server, helpers.Client)
+			cmd := fmt.Sprintf("super_netperf %d -t TCP_RR -H %s", num, serverIP)
+			if client {
+				res = vm.ContainerExec(helpers.Client, cmd)
+			} else {
+				res = vm.ContainerExec(helpers.Server, cmd)
+			}
+
+			res.ExpectSuccess()
+			fmt.Fprintf(&PerfLogWriter, "%s,", strings.TrimSuffix(res.GetStdOut(), "\n"))
+		}
+
+		SuperNetperfRR := func(client bool, num int, serverIP string, serverIPv6 string) {
+			SuperNetperfRRIPv4(client, num, serverIP)
+			ClientServerCheck(serverIPv6, serverIP)
+		}
+
+		SuperNetperfStreamIPv4 := func(client bool, num int, serverIP string) {
+			var res *helpers.CmdRes
+
+			By("super_netperf to %s from %s (should succeed)", helpers.Server, helpers.Client)
+			cmd := fmt.Sprintf("super_netperf %d -f g -t TCP_STREAM -H %s", num, serverIP)
+			if client {
+				res = vm.ContainerExec(helpers.Client, cmd)
+			} else {
+				res = vm.ContainerExec(helpers.Server, cmd)
+			}
+
+			res.ExpectSuccess()
+			fmt.Fprintf(&PerfLogWriter, "%s,", strings.TrimSuffix(res.GetStdOut(), "\n"))
+		}
+
+		SuperNetperfStream := func(client bool, num int, serverIP string, serverIPv6 string) {
+			SuperNetperfStreamIPv4(client, num, serverIP)
+			ClientServerCheck(serverIPv6, serverIP)
+		}
+
+		NetperfTest := func() {
```

Comment:
I'd recommend splitting up this entire test into separate `Its`, where each test runs a `SuperNetPerf` test in each `It`. This makes it so that each `It` corresponds to roughly one assertion as opposed to running everything within one `It`. You could get the `serverData` and `serverIPv6` information outside of the `It` (say in `BeforeAll`), and then have each `It` run the netperf tests.

Example:

```
It("L4 Netperf over loopback with 1 connection", func() {
    SuperNetPerfRR(false, 1, serverIP, serverIPv6)
})
```

@eloycoto please confirm what I am saying above, as I believe you have vocalized opinions that echo what I have stated before.

Reply:
Will the test_results get a bit noisy if we do this? I think we get a directory per It instance.

Reply:
@jrfastab do as Ian mentionend. The test results does not matter at all because the output will be  attached in the Jenkins so it's easy   ;-)

Reply:
OK will do this today.